test/test_s3_utils.py

```diff
@@ -269,66 +270,136 @@ def test_s3utils_creation_cgap_stg():
     assert compute_cgap_stg_env() is None, "There seems to be a CGAP staging environment. Tests need updating."
 
 
+def _check_portal_auth(*, portal_env: str, auth_getter: Callable, auth_kind: str, server_pattern: str,
+                       other_getters: Optional[Dict[str, Optional[Callable]]] = None, require_key=True):
```

Comment:
This checker function is new and shared by a number of the other tests to avoid repetition.